### PR TITLE
CAF-3567: New output mode added for language detection result, output…

### PR DESCRIPTION
**Pull request by mmaBel**
…ting a stringified object.

### CODE REVIEW

File: worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionResultFormat.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            enumNamesToValues.put(enumValue.name().toLowerCase(Locale.ENGLISH), enumValue);
+        }
+    }
+
+    /**
+     * Retrieve enum value the provided string value represents.
+     * @param valueAsStr name of enum value
+     * @return the enum value matching provided string value or null if there is no match
+     */
+    public static LanguageDetectionResultFormat tryGetValueOf(String valueAsStr)
```
Comment (kploch) [inline worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionResultFormat.java]: Why did you not use the valueOf? If the case is the problem then you could call toUpperCase() on the string value...

File: worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionUtilities.java [hunk 2]
```diff
@@ -159,24 +154,97 @@ public static void outputDocumentFieldValueChanges(final Document document)
...
+
         // Add detected languages to the document object.
-        if (inMultiFeildMode) {
-            LOG.debug("Adding metadata to the document for each language detected.");
-            addDetectedLanguagesToDocument(detectorResult, document, sourceDataField.getName());
-        } else {
-            // Add detected languages to the document object.
-            LOG.debug("Adding metadata to the document for each language detected.");
-            addDetectedLanguagesToDocument(detectorResult, document);
+        if(resultFormat.equals(LanguageDetectionResultFormat.SIMPLE))
```
Comment (kploch) [inline worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionUtilities.java]: You can compare enum values with ==

File: worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionResultFormat.java (hunk 1 above)
Comment (mmaBel) [reply worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionResultFormat.java]: Enum valueOf throws Illegal argument exception if the passed string doesn't match. I prefer this method over a try/catch and case change in the method using the enum.

File: worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionUtilities.java (hunk 2 above)
Comment (mmaBel) [reply worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionUtilities.java]: Updated to ==.

File: worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionWorker.java [hunk 3]
```diff
@@ -153,11 +165,29 @@ private void detectLanguage(final Document document, final String fieldName, fin
...
+     */
+    private LanguageDetectionResultFormat getResultFormatToUse(Document document)
+    {
+        final String resultFormatStr = document.getCustomData(LanguageDetectionConstants.CustomData.RESULT_FORMAT);
+        if(Strings.isNullOrEmpty(resultFormatStr)){
+            return LanguageDetectionResultFormat.SIMPLE;
+        }
+        else {
+            LanguageDetectionResultFormat convertedResultFormat =
+                    LanguageDetectionResultFormat.tryGetValueOf(resultFormatStr);
```
Comment (kploch) [inline worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionWorker.java]: This is similar to my previous comment about having tryGetValueOf - if there's an invalid value, we should fail. I looked into the language detection worker and there's quite a lot of bad exception handling so in this case we should: use the valueOf method and catch the Invalid argument exception thrown (it's unfortunate that it's a runtime exception but we should add in this case an error to the document).

Comment (mmaBel) [reply worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionWorker.java]: OK.

File: worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionResultFormat.java (hunk 1 above)
Comment (mmaBel) [reply worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionResultFormat.java]: Changed to use valueOf and throw RuntimeException if invalid value provided.

**Pull request merged**